openquake/hazardlib/gsim/parker_2020.py

```diff
@@ -0,0 +1,575 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with OpenQuake. If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Module exports :class:`ParkerEtAl2020SInter`
+               :class:`ParkerEtAl2020SInterB`
+               :class:`ParkerEtAl2020SSlab`
+               :class:`ParkerEtAl2020SSlabB`
+"""
+import math
+
+import numpy as np
+from scipy.special import erf
+
+from openquake.hazardlib import const
+from openquake.hazardlib.gsim.base import GMPE, CoeffsTable, gsim_aliases
+from openquake.hazardlib.imt import PGA, SA, PGV
+
+
+class ParkerEtAl2020SInter(GMPE):
+    """
+    Implements Parker et al. (2020) for subduction interface.
+    """
+
+    DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.SUBDUCTION_INTERFACE
+
+    #: Supported intensity measure types are spectral acceleration,
+    #: peak ground acceleration and peak ground velocity
+    DEFINED_FOR_INTENSITY_MEASURE_TYPES = set([
+        PGV,
+        PGA,
+        SA
+    ])
+
+    #: Supported intensity measure component is the geometric mean component
+    DEFINED_FOR_INTENSITY_MEASURE_COMPONENT = const.IMC.AVERAGE_HORIZONTAL
+
+    DEFINED_FOR_STANDARD_DEVIATION_TYPES = set([
+        const.StdDev.TOTAL,
+        const.StdDev.INTER_EVENT,
+        const.StdDev.INTRA_EVENT
+    ])
+
+    #: Site amplification is dependent only upon Vs30
+    REQUIRES_SITES_PARAMETERS = {'vs30'}
+
+    #: Required rupture parameters are only magnitude for the interface model
+    REQUIRES_RUPTURE_PARAMETERS = {'mag'}
+
+    #: Required distance measure is closest distance to rupture, for
+    #: interface events
+    REQUIRES_DISTANCES = {'rrup'}
+
+    def __init__(self, region=None, saturation_region=None, basin=None):
+        """
+        Enable setting regions to prevent messy overriding
+        and code duplication.
+        """
+        super().__init__(region=region, saturation_region=saturation_region,
+                         basin=basin)
+        self.region = region
+        if saturation_region is None:
+            self.saturation_region = region
+        else:
+            self.saturation_region = saturation_region
+        self.basin = basin
+
+    def get_mean_and_stddevs(self, sites, rup, dists, imt, stddev_types):
+        """
+        See :meth:`superclass method
+        <.base.GroundShakingIntensityModel.get_mean_and_stddevs>`
+        for spec of input and result values.
+        """
+        # extract dictionaries of coefficients specific to required
+        # intensity measure type and for PGA
+        C = self.COEFFS[imt]
+        C_PGA = self.COEFFS[PGA()]
+
+        m_b = self._get_mb()
+        c0, c0_pga = self._c0(C, C_PGA)
+        fm, fm_pga = self._magnitude_scaling(C, C_PGA, rup.mag, m_b)
+        fp, fp_pga = self._path_term(C, C_PGA, rup.mag, dists.rrup, m_b)
+        if hasattr(rup, 'hypo_depth'):
```

Comment:
After checking the SSLab part I understand why you added this here. It's a bit borderline.

Reply:
Wasn't sure if I should change it to a parameter/init kwarg or duplicate this part in sslab.

Reply:
I've cleaned it up a little.